src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,221 @@
+                + "WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement updateStatement = connection.prepareStatement(
+                         updateRequest)) {
+            updateStatement.setString(1, car.getModel());
+            updateStatement.setLong(2, car.getManufacturer().getId());
+            updateStatement.setLong(3, car.getId());
+            updateStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update a car " + car, e);
+        }
+        deleteDrivers(car.getId());
+        addDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteRequest = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement deleteStatement = connection.prepareStatement(
+                         deleteRequest)) {
+            deleteStatement.setLong(1, id);
+            return deleteStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete a car by id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getAllByDriverRequest = "SELECT c.id, c.manufacturer_id, c.model, m.name, m.country "
+                + "FROM cars AS c "
+                + "INNER JOIN manufacturers AS m "
+                + "ON c.manufacturer_id = m.id "
+                + "INNER JOIN cars_drivers AS cd "
+                + "ON c.id = cd.car_id "
+                + "WHERE cd.driver_id - ? "
+                + "AND c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getAllByDriverStatement = connection.prepareStatement(
+                         getAllByDriverRequest)) {
+            getAllByDriverStatement.setLong(1, driverId);
+            ResultSet resultSet = getAllByDriverStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+            for (Car car : cars) {
+                car.setDrivers(getDrivers(car.getId()));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't delete all cars by a driver's id " + driverId, e);
+        }
+    }
+
+    private void addDrivers(Car car) {
+        String addDriversRequest = "INSERT INTO cars_drivers (driver_id, car_id) "
+                + "VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement addDriversStatement = connection.prepareStatement(
+                         addDriversRequest, Statement.RETURN_GENERATED_KEYS)) {

```

Comment:
```suggestion
                         addDriversRequest)) {
```
тут ід не генерується